packages/mask/src/extension/popups/modals/WalletAutoLockSettingModal/index.tsx

```diff
@@ -0,0 +1,185 @@
+import { forwardRef, useState } from 'react'
+import { useAsyncFn } from 'react-use'
+import millisecondsToMinutes from 'date-fns/millisecondsToMinutes'
+import minutesToMilliseconds from 'date-fns/minutesToMilliseconds'
+import { Box, TextField, Typography, useTheme } from '@mui/material'
+import { CrossIsolationMessages, type SingletonModalRefCreator } from '@masknet/shared-base'
+import { ActionButton, makeStyles } from '@masknet/theme'
+import { useSingletonModal } from '@masknet/shared-base-ui'
+import { WalletRPC } from '../../../../plugins/WalletService/messages.js'
+import { BottomDrawer, type BottomDrawerProps } from '../../components/index.js'
+import { useI18N } from '../../../../utils/i18n-next-ui.js'
+import { useWalletAutoLockTime } from '../../pages/Wallet/hooks/useWalletAutoLockTime.js'
+
+const useStyles = makeStyles()((theme) => ({
+    list: {
+        display: 'flex',
+        justifyContent: 'space-between',
+        alignItems: 'flex-start',
+        marginTop: 16,
+        gap: 12,
+        alignSelf: 'stretch',
+    },
+    listItem: {
+        padding: '8px 12px',
+        borderRadius: 6,
+        display: 'flex',
+        gap: 4,
+        justifyContent: 'center',
+        alignItems: 'center',
+        height: 32,
+        background: theme.palette.maskColor.thirdMain,
+        flex: '1 0 0',
+        fontSize: 12,
+        fontWeight: 700,
+        cursor: 'pointer',
+    },
+    selected: {
+        color: theme.palette.maskColor.bottom,
+        background: theme.palette.maskColor.main,
+    },
+    listItemOneDay: {
+        flex: 'auto',
+        width: 48,
+    },
+}))
+
+enum OptionName {
+    ONE_QUARTE = '15',
+    TWO_QUARTE = '30',
+    ONE_HOUR = '60',
+    ONE_DAY = '1 Day',
+    NEVER = 'Never',
+}
+
+const DEFAULT_MIN_AUTO_LOCKER_TIME = 1000 * 60 * 15 // 15 mins
+
+function WalletAutoLockSettingDrawer(props: BottomDrawerProps) {
+    const { t } = useI18N()
+    const theme = useTheme()
+    const { classes, cx } = useStyles()
+    const { value: autoLockerTime = DEFAULT_MIN_AUTO_LOCKER_TIME } = useWalletAutoLockTime()
+
+    const [time, setTime] = useState<string>()
+
+    const initialTime = millisecondsToMinutes(autoLockerTime).toString()
+
+    const error = Number.isNaN(Number(time ?? initialTime))
```

Comment:
Use `date-fns` to parse a timestamp.

Reply:
This is not a timestamp. But to require number only input.